src/components/common/incomplete-tests-banner/incomplete-tests-banner.ts

```diff
@@ -34,17 +35,17 @@ export class IncompleteTestsBanner implements OnInit {
   constructor(
     private store$: Store<StoreModel>,
     private slotProvider: SlotProvider,
-  ) {}
+    private dateTimeProvider: DateTimeProvider,
+  ) {
+  }
 
   ngOnInit() {
     this.componentState = {
       count$: this.store$.pipe(
         select(getJournalState),
-        withLatestFrom(this.store$.pipe(
-          select(getTests),
-        )),
+        withLatestFrom(this.store$.pipe(select(getTests))),
         map(([journal, tests]) =>
-          getIncompleteTestsCount(journal, tests, this.todaysDate, this.slotProvider)),
+          getIncompleteTests(journal, tests, this.dateTimeProvider.now(), this.slotProvider).length),
       ),
```

Comment:
We should house all this logic inside the unuploaded test selector file instead of copying these steps each time.